src/Shape_buildblock/GenerateImage.cxx

```diff
@@ -204,6 +232,29 @@ post_processing()
     warning("number of samples to take in x-direction should be strictly positive\n");
     return true;
   }
+
+  tmpl_image.reset( new VoxelsOnCartesianGrid<float>(exam_info_sptr,
+                                                    IndexRange3D(0,output_image_size_z-1,
+                                                                 -(output_image_size_y/2),
+                                                                 -(output_image_size_y/2)+output_image_size_y-1,
+                                                                 -(output_image_size_x/2),
+                                                                 -(output_image_size_x/2)+output_image_size_x-1),
+                                                    CartesianCoordinate3D<float>(0,0,0),
+                                                    CartesianCoordinate3D<float>(output_voxel_size_z,
+                                                                                 output_voxel_size_y,
+                                                                                 output_voxel_size_x)));
+
+  shared_ptr<Scanner> scn(Scanner::get_scanner_from_name((exam_info_sptr->originating_system)));
+  out_density_ptr.reset(new DynamicDiscretisedDensity(exam_info_sptr->get_time_frame_definitions(),
+                                                      rel_start_time,
+                                                      scn,
+                                                      tmpl_image) );
+
```

Comment:
I'm astonished that we don't have a constructor for `DynamicDiscretisedDensity` that takes an `ExamInfo`. That would be far better. Maybe add it in this PR?